Lib/Protocols/IdFTP.pas

```diff
@@ -2742,18 +2771,27 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
-
+  if IsExtSupported('CSID') then begin {do not localize}
+      LClnt := FClientInfo.GetCSIDOutput;
+      SendCmd('CSID ' + LClnt);  {do not localize}
+      if Self.LastCmdResult.NumericCode = 200 then begin
+        LBuf := TrimRight(Self.LastCmdResult.Text.Text);
+        FServerInfo.ServerInfo.Clear;

```

Comment:
The `ServerInfo` should be cleared before sending the `CSID` command. Or at least also cleared if the command fails.